api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -516,6 +523,37 @@ message HttpConnectionManager {
   bool strip_matching_host_port = 39;
 }
 
+// The configuration to customize local reply returned by Envoy.
+message LocalReplyConfig {
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The mappers will be checked by the specified order until one is matched.
+  repeated ResponseMapper mappers = 1;
+
+  // The configuration to form response body from the :ref:`command operators <config_access_log_command_operators>`
+  // and to specify response content type as one of: plain/text or application/json.
+  // If the matched `mapper <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.LocalReplyConfig.mapper>`
+  // has a valid `format <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.ResponseMapper.body_format>`,
+  // it will be used, this one will not be used.
+  config.core.v3.SubstitutionFormatString body_format = 2;
+}
+
+// The configuration to filter and change local response.
+message ResponseMapper {
+  // Filter to determine if this mapper should apply.
+  config.accesslog.v3.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
+
+  // The new response status code if specified.
+  google.protobuf.UInt32Value status_code = 2 [(validate.rules).uint32 = {lt: 600 gte: 200}];
+
+  // The new response body if specified.
+  config.core.v3.DataSource body = 3;
```

Comment:
Is this `oneof` with `body_format`?

Reply:
no,  this is customized local reply body text.  at the same time, users can also specify body_format.